lib/reservation_manager.rb

```diff
@@ -0,0 +1,126 @@
+require_relative "reservation"
+
+class Reservation_manager
+  attr_reader :reservations, :pending_reservations_for_blocks
+
+  def initialize
+    @reservations = []
+    @pending_reservations_for_blocks = []
+  end
+
+  def all_rooms
+    @all_rooms = []
+    20.times do |number|
+      @all_rooms << number + 1
+    end
+    return @all_rooms
+  end
+
+  def find_available_rooms(check_in_day, check_out_day)
+    available_rooms = (1..20).to_a
+    check_in = Date.parse(check_in_day)
+    check_out = Date.parse(check_out_day)
+
+    date_range_of_interest = (check_in...check_out).to_a
+
+    @reservations.each do |reservation|
+      date_range = (reservation.check_in_day...reservation.check_out_day).to_a
+      combined_range = date_range + date_range_of_interest
+
+      if combined_range.length != combined_range.uniq.length
```

Comment:
It might be interesting if `combined_range.length != combined_range.uniq.length` and the logic above it (starting with `date_range = ...`) were in a helper method that took in a `reservation` and returned a boolean